controller_manager:
  ros__parameters:
    update_rate: 1000
    use_sim_time: true

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    joint_group_position_controller:
      type: position_controllers/JointGroupPositionController

joint_group_position_controller:
  ros__parameters:
    joints:
      - joint1
      - joint2
    
    command_interfaces:
      - position
    
    state_interfaces:
      - position
      - velocity
      - effort
    
    state_publish_rate: 1000.0

    gains:
      joint1:
        p: 100.0
        i: 0.01
        d: 5.0
      joint2:
        p: 80.0
        i: 0.01
        d: 4.0